aptos-move/block-executor/src/scheduler_status.rs

```diff
@@ -1149,114 +1241,145 @@ mod tests {
-                8,
+        for (status, txn_idx) in [
+            (
+                ExecutionStatus::new_for_test(
+                    InnerStatus::new_for_test(StatusEnum::Executed, 5),
+                    0,
+                ),
+                8u32,
             ),
-            ExecutionStatus::new_for_test(
-                InnerStatus::new_for_test(StatusEnum::Executed, 5),
-                0,
-                &manager,
-                13,
+            (
+                ExecutionStatus::new_for_test(
+                    InnerStatus::new_for_test(StatusEnum::Executed, 5),
+                    0,
+                ),
+                13u32,
             ),
-            ExecutionStatus::new_for_test(
-                InnerStatus::new_for_test(StatusEnum::Executed, 0),
-                0,
-                &manager,
-                10,
+            (
+                ExecutionStatus::new_for_test(
+                    InnerStatus::new_for_test(StatusEnum::Executed, 0),
+                    0,
+                ),
+                10u32,
             ),
         ] {
+            *statuses.get_status_mut(txn_idx) = status;
+            let status = &statuses.get_status(txn_idx);
             // Double-check expected state - provided by new_for_test
-            assert!(status.is_executed());
+            assert!(statuses.is_executed(txn_idx));
             assert_eq!(
                 status.dependency_shortcut.load(Ordering::Relaxed),
                 DependencyFlag::Safe as u8
             );
-            assert!(!status.pending_scheduling_and_not_stalled());
+            assert!(!statuses.pending_scheduling_and_not_stalled(txn_idx));
 
             let new_incarnation = status.next_incarnation_to_abort.load(Ordering::Relaxed) + 1;
-            status.incarnate(&mut status.inner_status.lock(), new_incarnation, false);
+            statuses.to_pending_scheduling(
+                txn_idx,
+                &mut status.inner_status.lock(),
+                new_incarnation,
+                add_to_schedule,
+            );
 
             assert_eq!(status.inner_status.lock().incarnation(), new_incarnation);
             assert_eq!(
                 status.dependency_shortcut.load(Ordering::Relaxed),
                 DependencyFlag::Defer as u8
             );
             assert_eq!(
-                status.pending_scheduling_and_not_stalled(),
+                statuses.pending_scheduling_and_not_stalled(txn_idx),
                 !status.is_stalled()
             );
 
-            expected_queue.push(status.txn_idx);
-            manager.assert_execution_queue(&expected_queue);
+            if add_to_schedule {
+                expected_queue.push(txn_idx);
+            }
+            statuses
+                .get_execution_queue_manager()
+                .assert_execution_queue(&expected_queue);
         }
     }
 
     #[test_case(false)]
     #[test_case(true)]
     fn assert_finish_execution_status(case: bool) {
```

Comment:
nit: should we have a better name rather than just "case"?

Reply:
done